code/game/objects/items/shooting_range.dm

```diff
@@ -1,76 +1,120 @@
+#define TARGETTING_DUMMY_USE_DELAY 20
+#define TARGETTING_DUMMY_WELD_DELAY 10
+
 // Targets, the things that actually get shot!
-/obj/item/target
+/obj/item/target/
 	name = "shooting target"
 	desc = "A shooting target."
 	icon = 'icons/obj/objects.dmi'
 	icon_state = "target_h"
 	density = FALSE
 	resistance_flags = INDESTRUCTIBLE
 
-	Destroy()
-		// if a target is deleted and associated with a stake, force stake to forget
-		for(var/obj/structure/target_stake/T in view(3,src))
-			if(T.pinned_target == src)
-				T.pinned_target = null
-				T.density = TRUE
-				break
-		. = ..() // delete target
-
-	Move()
-		..()
-		// After target moves, check for nearby stakes. If associated, move to target
-		for(var/obj/structure/target_stake/M in view(3,src))
-			if(M.density == 0 && M.pinned_target == src)
-				M.loc = loc
-
-		// This may seem a little counter-intuitive but I assure you that's for a purpose.
-		// Stakes are the ones that carry targets, yes, but in the stake code we set
-		// a stake's density to 0 meaning it can't be pushed anymore. Instead of pushing
-		// the stake now, we have to push the target.
+/obj/item/target/default
+	desc = "A shooting target with a distinctly human outline."
 
+/obj/item/target/syndicate
+	icon_state = "target_s"
+	desc = "A shooting target that looks like a hostile agent."
+	
+/obj/item/target/alien
+	icon_state = "target_q"
+	desc = "A shooting target with a threatening silhouette."
+	
+///Basically these are for the firing range
+/obj/structure/target_stake
+	name = "target stake"
+	desc = "A thin platform with negatively-magnetized wheels."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "target_stake"
+	density = TRUE
+	flags_atom = CONDUCT
+	max_integrity = 15000 //important that what the marines are shooting at doesn't break, we don't make it invulnerable because we still need to plasma cutter it sometimes
+	soft_armor = list("melee" = 80, "bullet" = 100, "laser" = 100, "energy" = 100, "bomb" = 120, "bio" = 100, "rad" = 0, "fire" = 100, "acid" = 0)
 
+/obj/structure/target_stake/attackby(obj/item/I, mob/user, params)
+	. = ..()		
+	if(istype(I, /obj/item/target))
+		var/obj/item/target/targetcushion = I
+		to_chat(user, "You start fitting the target onto the stake.")
+		if(!do_after(user, TARGETTING_DUMMY_USE_DELAY, TRUE, src, BUSY_ICON_FRIENDLY))
```

Comment:
early return

Reply:
theres one istype, then a massive block of indented code with a return at the end, both the ending return and the massive indentation is uneeded